src/main/java/ca/ciccc/MagicSquare.java

```diff
@@ -71,8 +71,47 @@ public void setSquare(Integer[][] square) {
      */
     public static boolean isMagicSquare(Integer[][] square) {
         // TODO: You need to implement this method.
+        int answer = 0;
+        int rowTotal = square.length;
 
-        return false;
+        for(int i = 0; i < rowTotal; i++ ){
+            int case1 = 0;
+
+            for(int p = 0; p < rowTotal; p++){
+                case1 = case1 + square[p][i];
+
+            }
+
+            answer = case1;
+        }
+
+        for(int i = 0; i < rowTotal; i++){
+                int case2 = 0;
+
+            for(int p = 0; p < rowTotal; p++){
+                case2 = case2 + square[p][i];
+
+            }
+
+            if(case2 != answer){
+                return false;
+            }
+        }
+
+        for(int i =0; i < rowTotal; i++){
```

Comment:
You also need to evaluate sum of each array. I think you meant case 3 is 
case3 = case3 + square[i][p];
then make sense.